Scripts/Runtime/Job/AccessControl/WrappedAccessControlledValue.cs

```diff
@@ -0,0 +1,148 @@
+using Anvil.CSharp.Core;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Jobs
+{
+    /// <summary>
+    /// Wraps an <see cref="AccessControlledValue{T}"/> and allows the controlled value to be transformed
+    /// before delivering a <see cref="TWrapped"/> instance to the consumer.
+    ///
+    /// This is an effective way of providing a constrained view of a broader internal access controlled value.
+    /// Ex: A reader/writer to an internal stream.
+    /// </summary>
+    /// <typeparam name="TWrapped">The type of the wrapped instance delivered to the consumer.</typeparam>
+    /// <typeparam name="TBase">The base access controlled value that is being wrapped.</typeparam>
+    public class WrappedAccessControlledValue<TWrapped, TBase> : AbstractAnvilBase,
+                                                                 IAccessControlledValue<TWrapped>,
+                                                                 IReadAccessControlledValue<TWrapped>,
+                                                                 ISharedWriteAccessControlledValue<TWrapped>,
+                                                                 IExclusiveWriteAccessControlledValue<TWrapped>
+    {
+        /// <summary>
+        /// Given the base value <see cref="TBase"/> produces the <see cref="TWrapped"/> to deliver to the consumer
+        /// of this instance.
+        /// </summary>
+        public delegate TWrapped WrapBaseValueDelegate(TBase baseValue);
+
+        private readonly AccessControlledValue<TBase> m_BaseACV;
+        private readonly WrapBaseValueDelegate m_WrapBaseValue;
+
+        /// <summary>
+        /// Creates a new instance of a <see cref="WrappedAccessControlledValue{TWrapped,TBase}"/>.
+        /// </summary>
+        /// <param name="baseACV">The base <see cref="AccessControlledValue{T}"/> to drive access.</param>
+        /// <param name="wrapBaseValue">Given the base value produces the wrapper value to deliver to the consumer.</param>
+        public WrappedAccessControlledValue(AccessControlledValue<TBase> baseACV, WrapBaseValueDelegate wrapBaseValue)
```

Comment:
I think using the delegate is completely fine in this case but just bringing it up for discussion. 

The other pattern i've seen is the use of a converter class like here: https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.comparer-1?view=net-8.0

So we would have `WrappedAccessControlledValue<TWrapped, TBase, TWrapper<TBase, TWrapper>> where TWrapper : IWrapper<TBase, TWrapper>`

You wouldn't need to create a lambda because we would instantiate the instance of the IWrapper inside here and just pass it through the concrete method of converting a TBase to a TWrapped. 

It's definitely better performance wise, but how much better I have no idea. 

The additional benefit is that if you have `WrappedAccessControlledValue`'s located in different parts of the app that both need to convert the same TBase to TWrapped, you don't need to manually make sure your lambdas are the same. You can have that logic consistent in the same place. 

This could also just be a TODO or Github issue.

Reply:
That's an interesting solution but I'm not sure it works well for this.
 - We'd want to pass in the converter instance so that the developer can pull in other dependencies to construct the wrapper if required.
 - The boilerplate required for a separate converter class starts to diminish the value of the wrapper to begin with. We're not much less code then creating the boilerplate required without the wrapper.

If we're concerned about performance we may as well just make `WrappedAccessControlledValue` abstract and have developers subclass it to define their wrap creator logic.

Happy to discuss though if you still see a case.

Reply:
Ok those are fair counter-points. The main one winning me over is the need to pull in other dependencies. 

We can always revisit if we need to and it becomes a problem.